src/kbmod/regionsearch/indexers.py

```diff
@@ -0,0 +1,70 @@
+"""
+lincc-frameworks provides implementations of ``abstractions.ObserverationIndexer`` that may be composed with an implementation of ``abstractions.Backend`` to provide a complete region search.
+"""
+import math
+
+import numpy
+from astropy import coordinates
+from astropy import units as units
+from astropy.coordinates import EarthLocation, SkyCoord, solar_system_ephemeris
+from astropy.time import Time
+
+from kbmod.regionsearch.abstractions import ObservationIndexer
+
+
+class PartitionIndexer(ObservationIndexer):
+    """Assigns a provided index to observation volumes that intersect the configured space and a second index to those that do not.
+    ."""
+
+    def __init__(
+        self,
+        search_ra: coordinates.Longitude,
+        search_dec: coordinates.Latitude,
+        search_distance: coordinates.Distance,
+        search_fov: coordinates.Angle,
+        is_in_index: int,
+        is_out_index: int,
+        **kwargs
+    ):
+        """Initializes the PartitionIndexer object.
+
+        Parameters
+        ----------
+
+        search_ra : astropy.coordinates.Longitude
+            The right ascension of the center of the sphere.
+        search_dec : astropy.coordinates.Latitude
+            The declination of the center of the sphere.
+        search_distance : astropy.coordinates.Distance
+            The distance from the barycenter to the center of the sphere.
+        search_fov : astropy.coordinates.Angle
+            The angle subtended by the sphere at the barycenter.
+        is_in_index : int
+            The index to assign to observations that intersect the sphere.
+        is_out_index : int
+            The index to assign to observations that do not intersect the sphere.
+        """
+        super().__init__(**kwargs)
+        self.obssky = coordinates.SkyCoord(search_ra, search_dec, distance=search_distance, frame="icrs")
+        self.fov = search_fov
+        self.radius = math.sin(search_fov.to(units.rad).value / 2) * search_distance
+        self.is_in_index = is_in_index
+        self.is_out_index = is_out_index
+
+    def observations_to_indices(
+        self, pointing: SkyCoord, time: Time, fov: coordinates.Angle, location: EarthLocation
+    ) -> numpy.ndarray:
+        """Returns a list of indices"""
+        if time is None:
+            separation = self.obssky.separation(pointing)
+        else:
+            if location is None:
+                location = time.location
+
+            with solar_system_ephemeris.set("de432s"):
```

Comment:
or I could remove the fallback to a location given in the time if that is less confusing. The preferred use in the location as a separate column and for many time columns no location is set (in which case this changes nothing).

Reply:
I actually meant the coordinate transforms: ITRS -> GCRS.

Reply:
I asked Dino to take a look

Reply:
`EarthLocation` has a `get_gcrs` so there's no need to get it as ITRS in the first point? See [documentation](https://docs.astropy.org/en/stable/api/astropy.coordinates.EarthLocation.html#astropy.coordinates.EarthLocation.get_gcrs). 

Overall I'm not sure I understand what the method is doing, the docs are a bit stiff and there are some oddities. Work in progress, I'm sure. This should work, but I'd recommend just getting a GCRS coordinate. 

I will just drop a few notes that "radius" instead of "fov" is the more usual name (search within this radius). FOV could be a solid angle and just generally a bit confusing. 

You want to transform to cartesian system and then search a box because `.separation` is the Haversine distance and is very costly. See how I did it in the [example implementation](https://github.com/DinoBektesevic/region_search_example/blob/main/models.py#L261). This is why this same approach works for 80 mil rows for CADC, you round all that down to integers, execute a fast select on indexed integers in the DB, and then you prune on finer shapes. You might not intending for this function to do that so this might be fine, or not, but if you were this won't be fast if there are many paintings.

`Time.location` is not guaranteed not to be `None`, I'd rather have that a mandatory parameter.

The `ephemeris.set` only works if there's internet - so this is in general something that either has to be clear to the user and optionally configurable. On Hyak worker nodes, for example, you can't reach outside internet.